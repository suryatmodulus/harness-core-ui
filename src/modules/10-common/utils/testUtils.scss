/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

.browser {
  border: 1px solid #cccccc;
  border-radius: 8px;
  min-height: 300px;
  overflow: auto;

  .browser-header {
    background-color: #cccccc;
    padding: 8px 16px;
    display: flex;
  }

  .browser-path {
    display: block;
    border: none;
    line-height: 14px;
    font-size: 12px;
    color: #444444;
    border-radius: 9px;
    min-width: 100px;
    padding: 2px 8px;
    flex: 1 1 auto;
    border: 1px solid transparent;

    &:focus {
      border-color: rgb(30, 167, 253);
      outline: none;
    }
  }

  .browser-content {
    padding: 8px;
  }
}
