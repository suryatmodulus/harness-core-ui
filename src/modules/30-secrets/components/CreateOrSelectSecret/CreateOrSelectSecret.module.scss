.main {
  padding: var(--spacing-xlarge) var(--spacing-xlarge);
  position: relative;
  top: -50px;
  :global {
    .bp3-tab-indicator-wrapper .bp3-tab-indicator {
      background-color: var(--primary-7);
      height: 3.5px;
    }
  }
}
