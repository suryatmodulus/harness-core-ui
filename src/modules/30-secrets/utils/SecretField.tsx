/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import { Scope } from '@common/interfaces/SecretsInterface'
import { getSecretV2Promise, GetSecretV2QueryParams } from 'services/cd-ng'

export interface SecretReferenceInterface {
  identifier: string
  name: string
  referenceString: string
  accountIdentifier: string
  orgIdentifier?: string
  projectIdentifier?: string
}

export const setSecretField = async (
  secretString: string,
  scopeQueryParams: GetSecretV2QueryParams
): Promise<SecretReferenceInterface | void> => {
  if (!secretString) {
    return undefined
  } else {
    const secretScope = secretString?.indexOf('.') < 0 ? Scope.PROJECT : secretString?.split('.')[0]

    switch (secretScope) {
      case Scope.ACCOUNT:
        delete scopeQueryParams.orgIdentifier
        delete scopeQueryParams.projectIdentifier
        break
      case Scope.ORG:
        delete scopeQueryParams.projectIdentifier
    }

    const identifier = secretString.indexOf('.') < 0 ? secretString : secretString.split('.')[1]
    const response = await getSecretV2Promise({
      identifier,
      queryParams: scopeQueryParams
    })

    return {
      identifier,
      name: response.data?.secret.name || secretString.split('.')[1],
      referenceString: secretString,
      ...scopeQueryParams
    }
  }
}
