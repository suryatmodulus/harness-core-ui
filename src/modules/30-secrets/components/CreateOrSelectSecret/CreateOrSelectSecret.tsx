/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { useParams } from 'react-router-dom'
import { pick } from 'lodash-es'
import SecretReference from '@secrets/components/SecretReference/SecretReference'
import { getReference } from '@common/utils/utils'
import type { SecretResponseWrapper, ResponsePageSecretResponseWrapper, ConnectorInfoDTO } from 'services/cd-ng'
import type { ProjectPathProps } from '@common/interfaces/RouteInterfaces'
import css from './CreateOrSelectSecret.module.scss'

export interface SecretReference {
  name: string
  identifier: string
  orgIdentifier?: string
  projectIdentifier?: string
  referenceString: string
}

export interface CreateOrSelectSecretProps {
  type?: SecretResponseWrapper['secret']['type']
  onSuccess: (secret: SecretReference) => void
  secretsListMockData?: ResponsePageSecretResponseWrapper
  connectorTypeContext?: ConnectorInfoDTO['type']
  onCancel?: () => void
  handleInlineSSHSecretCreation: () => void
}

const CreateOrSelectSecret: React.FC<CreateOrSelectSecretProps> = ({
  type,
  onSuccess,
  secretsListMockData,
  connectorTypeContext,
  onCancel,
  handleInlineSSHSecretCreation
}) => {
  const { accountId, orgIdentifier, projectIdentifier } = useParams<ProjectPathProps>()

  return (
    <section className={css.main}>
      <SecretReference
        type={type}
        onCancel={onCancel}
        onSelect={data => {
          onSuccess({
            ...pick(data, ['name', 'identifier', 'orgIdentifier', 'projectIdentifier']),
            referenceString: getReference(data.scope, data.identifier) as string
          })
        }}
        accountIdentifier={accountId}
        orgIdentifier={orgIdentifier}
        projectIdentifier={projectIdentifier}
        mock={secretsListMockData}
        connectorTypeContext={connectorTypeContext}
        handleInlineSSHSecretCreation={handleInlineSSHSecretCreation}
      />
    </section>
  )
}

export default CreateOrSelectSecret
