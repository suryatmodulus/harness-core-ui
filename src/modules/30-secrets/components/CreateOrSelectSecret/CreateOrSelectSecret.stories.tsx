/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import type { Meta, Story } from '@storybook/react'

import { Container } from '@wings-software/uicore'
import { TestWrapper } from '@common/utils/testUtils'
import CreateOrSelectSecret, { CreateOrSelectSecretProps } from './CreateOrSelectSecret'

export default {
  title: 'Secrets / CreateOrSelectSecret',
  component: CreateOrSelectSecret
} as Meta

export const Basic: Story<CreateOrSelectSecretProps> = args => (
  <TestWrapper path="/account/:accountId" pathParams={{ accountId: 'dummy' }}>
    <Container width={400}>
      <CreateOrSelectSecret {...args} />
    </Container>
  </TestWrapper>
)

Basic.args = {
  type: 'SecretText'
}
