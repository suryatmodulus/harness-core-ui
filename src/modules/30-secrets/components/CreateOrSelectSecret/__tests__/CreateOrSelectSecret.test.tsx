/*
 * Copyright 2021 Harness Inc. All rights reserved.
 * Use of this source code is governed by the PolyForm Shield 1.0.0 license
 * that can be found in the licenses directory at the root of this repository, also available at
 * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
 */

import React from 'react'
import { noop } from 'lodash-es'
import { render, waitFor, fireEvent, act } from '@testing-library/react'
import { TestWrapper } from '@common/utils/testUtils'
import CreateOrSelectSecret from '../CreateOrSelectSecret'
import connectorsListMockData from './connectorsListMockdata.json'
import secretsListMockData from './secretsListMockData.json'
import connectorDetailsMockData from './getConnectorMock.json'

const successCallback = jest.fn()

jest.mock('services/cd-ng', () => ({
  ...(jest.requireActual('services/cd-ng') as any),
  useGetConnectorList: () => {
    return {
      data: connectorsListMockData,
      refetch: jest.fn()
    }
  },
  useGetConnector: () => {
    return {
      data: connectorDetailsMockData,
      refetch: jest.fn()
    }
  }
}))

describe('CreateOrSelectSecret', () => {
  test('render', async () => {
    const { container, getByText } = render(
      <TestWrapper path="/account/:accountId" pathParams={{ accountId: 'dummy' }}>
        <CreateOrSelectSecret
          type="SecretText"
          onSuccess={successCallback}
          secretsListMockData={secretsListMockData as any}
          handleInlineSSHSecretCreation={noop}
        />
      </TestWrapper>
    )

    await waitFor(() => {
      expect(getByText('demo-1')).toBeDefined()
    })
    expect(container).toMatchSnapshot()
    const $secret = getByText('demo-1')
    await act(async () => {
      fireEvent.click($secret)
    })

    const $applyBtn = getByText('entityReference.apply')
    await act(async () => {
      fireEvent.click($applyBtn)
    })

    expect(successCallback).toHaveBeenCalled()
  })
})
