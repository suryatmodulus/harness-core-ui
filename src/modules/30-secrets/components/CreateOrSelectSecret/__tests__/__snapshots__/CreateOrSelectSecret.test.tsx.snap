// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateOrSelectSecret render 1`] = `
<div>
  <section
    class="main"
  >
    <div
      class="StyledProps--font StyledProps--main secretRefContainer"
    >
      <div
        class="StyledProps--font StyledProps--main container"
      >
        <div
          class="tabsContainer"
        >
          <div
            class="bp3-tabs Tabs--main Tabs--horizontal"
          >
            <div
              class="bp3-tab-list"
              role="tablist"
            >
              <div
                class="bp3-tab-indicator-wrapper"
                style="height: 0px; transform: translateX(0px) translateY(0px); width: 0px;"
              >
                <div
                  class="bp3-tab-indicator"
                />
              </div>
              <div
                aria-controls="bp3-tab-panel_selectScope_account"
                aria-disabled="false"
                aria-expanded="true"
                aria-selected="true"
                class="bp3-tab"
                data-tab-id="account"
                id="bp3-tab-title_selectScope_account"
                role="tab"
                tabindex="0"
              >
                <div
                  class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-flex-start"
                >
                  <span
                    class="bp3-icon bp3-icon-layers StyledProps--main tabIcon"
                    icon="layers"
                  >
                    <svg
                      data-icon="layers"
                      height="14"
                      viewBox="0 0 16 16"
                      width="14"
                    >
                      <desc>
                        layers
                      </desc>
                      <path
                        d="M.55 4.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5a.998.998 0 00-.06-1.81L8.4.08a1.006 1.006 0 00-.79 0l-6.99 3a.992.992 0 00-.07 1.81zM15 10c-.16 0-.31.04-.45.11L8 13.38 1.45 10.1c-.14-.06-.29-.1-.45-.1-.55 0-1 .45-1 1 0 .39.23.73.55.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5c.32-.16.55-.5.55-.89 0-.55-.45-1-1-1zm0-3.5c-.16 0-.31.04-.45.11L8 9.88 1.45 6.61A.997.997 0 001 6.5c-.55 0-1 .45-1 1 0 .39.23.73.55.89l7 3.5c.14.07.29.11.45.11s.31-.04.45-.11l7-3.5c.32-.16.55-.5.55-.89 0-.55-.45-1-1-1z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </span>
                  <p
                    class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-variation-h6 StyledProps--font-weight-light"
                    style="--text-line-clamp: 1;"
                  >
                    account
                  </p>
                  
                </div>
              </div>
            </div>
            <div
              aria-hidden="false"
              aria-labelledby="bp3-tab-title_selectScope_account"
              class="bp3-tab-panel"
              id="bp3-tab-panel_selectScope_account"
              role="tabpanel"
            >
              <div
                class="Layout--vertical Layout--layout-spacing-large StyledProps--main"
              >
                <div
                  class="searchBox"
                >
                  <div
                    class="TextInput--main search"
                  >
                    <div
                      class="bp3-input-group"
                    >
                      <span
                        class="bp3-icon bp3-icon-search StyledProps--main"
                        icon="search"
                      >
                        <svg
                          data-icon="search"
                          height="12"
                          viewBox="0 0 16 16"
                          width="12"
                        >
                          <desc>
                            search
                          </desc>
                          <path
                            d="M15.55 13.43l-2.67-2.68a6.94 6.94 0 001.11-3.76c0-3.87-3.13-7-7-7s-7 3.13-7 7 3.13 7 7 7c1.39 0 2.68-.42 3.76-1.11l2.68 2.67a1.498 1.498 0 102.12-2.12zm-8.56-1.44c-2.76 0-5-2.24-5-5s2.24-5 5-5 5 2.24 5 5-2.24 5-5 5z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                      <input
                        class="bp3-input"
                        placeholder="search"
                        type="text"
                        value=""
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="referenceList"
                >
                  <div
                    class="Collapse--main collapseWrapper"
                  >
                    <div
                      class="StyledProps--font StyledProps--main CollapseHeader--main Collapse--header collapseHeader hideCollapseIcon"
                    >
                      <div
                        class="StyledProps--font StyledProps--main CollapseHeader--leftSection"
                      >
                        <span
                          class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                          data-icon="main-chevron-right"
                        >
                          <svg
                            height="12"
                            viewBox="0 0 8 14"
                            width="12"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                              fill="#4d5969"
                            />
                          </svg>
                        </span>
                        <div
                          class="collapeHeaderContent"
                        >
                          <div
                            class="Layout--horizontal StyledProps--main item StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-space-between"
                          >
                            <div
                              class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main leftInfo"
                            >
                              <span
                                class="bp3-icon StyledProps--main iconCheck"
                                data-icon="pipeline-approval"
                              >
                                <svg
                                  height="14"
                                  viewBox="0 0 19 17"
                                  width="14"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M18.81.669l.006.004.015.013a.5.5 0 01.062.7L6.479 16.375a.5.5 0 01-.709.062.52.52 0 01-.136-.176.523.523 0 01-.126-.095L.132 10.578a.5.5 0 01.005-.698l.017-.018a.5.5 0 01.707-.005l.007.007 5.233 5.367L18.107.735a.5.5 0 01.704-.066z"
                                    fill="currentColor"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                              <div
                                class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center"
                              >
                                <span
                                  class="bp3-icon StyledProps--main secretIcon"
                                  data-icon="text"
                                >
                                  <svg
                                    fill="none"
                                    height="24"
                                    viewBox="0 0 16 19"
                                    width="24"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M11.585 18.334H4.467v-.667c.371-.135 1.916-1.009 2.356-1.076V1.667H2.947L1.3 5.14H.667c-.051-.338 0-4.469 0-4.807h14.718c0 .338.05 4.452 0 4.807h-.658L13.13 1.667H9.255V16.59c.439.084 1.976.924 2.33 1.076v.666z"
                                      fill="#3E7BFA"
                                    />
                                  </svg>
                                </span>
                                <div
                                  class="Layout--vertical StyledProps--main StyledProps--padding-left-small"
                                >
                                  <p
                                    class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-weight-bold StyledProps--color StyledProps--color-black"
                                    style="--text-line-clamp: 1;"
                                  >
                                    a1
                                  </p>
                                  <p
                                    class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-size-small StyledProps--font-weight-light StyledProps--color StyledProps--color-grey600"
                                    style="--text-line-clamp: 1;"
                                  >
                                    common.ID: a1.harnessSecretManager
                                  </p>
                                </div>
                              </div>
                            </div>
                            <button
                              class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main editBtn Button--with-current-color Button--without-shadow"
                              data-testid="-edit"
                              type="button"
                            >
                              <span
                                class="bp3-button-text"
                              >
                                <span
                                  class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey600"
                                  data-icon="Edit"
                                >
                                  <svg
                                    height="16"
                                    viewBox="0 0 16 16"
                                    width="16"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      clip-rule="evenodd"
                                      d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                                      fill="currentColor"
                                      fill-rule="evenodd"
                                    />
                                    <path
                                      clip-rule="evenodd"
                                      d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                                      fill="currentColor"
                                      fill-rule="evenodd"
                                      opacity="0.3"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                      <div
                        class="StyledProps--font StyledProps--main"
                      />
                    </div>
                    <div
                      class="Collapse--collapse"
                    >
                      <div
                        class="bp3-collapse"
                      >
                        <div
                          aria-hidden="false"
                          class="bp3-collapse-body"
                          style="transform: translateY(-undefinedpx);"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="Collapse--main collapseWrapper"
                  >
                    <div
                      class="StyledProps--font StyledProps--main CollapseHeader--main Collapse--header collapseHeader hideCollapseIcon"
                    >
                      <div
                        class="StyledProps--font StyledProps--main CollapseHeader--leftSection"
                      >
                        <span
                          class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                          data-icon="main-chevron-right"
                        >
                          <svg
                            height="12"
                            viewBox="0 0 8 14"
                            width="12"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                              fill="#4d5969"
                            />
                          </svg>
                        </span>
                        <div
                          class="collapeHeaderContent"
                        >
                          <div
                            class="Layout--horizontal StyledProps--main item StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-space-between"
                          >
                            <div
                              class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main leftInfo"
                            >
                              <span
                                class="bp3-icon StyledProps--main iconCheck"
                                data-icon="pipeline-approval"
                              >
                                <svg
                                  height="14"
                                  viewBox="0 0 19 17"
                                  width="14"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M18.81.669l.006.004.015.013a.5.5 0 01.062.7L6.479 16.375a.5.5 0 01-.709.062.52.52 0 01-.136-.176.523.523 0 01-.126-.095L.132 10.578a.5.5 0 01.005-.698l.017-.018a.5.5 0 01.707-.005l.007.007 5.233 5.367L18.107.735a.5.5 0 01.704-.066z"
                                    fill="currentColor"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                              <div
                                class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center"
                              >
                                <span
                                  class="bp3-icon StyledProps--main secretIcon"
                                  data-icon="text"
                                >
                                  <svg
                                    fill="none"
                                    height="24"
                                    viewBox="0 0 16 19"
                                    width="24"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M11.585 18.334H4.467v-.667c.371-.135 1.916-1.009 2.356-1.076V1.667H2.947L1.3 5.14H.667c-.051-.338 0-4.469 0-4.807h14.718c0 .338.05 4.452 0 4.807h-.658L13.13 1.667H9.255V16.59c.439.084 1.976.924 2.33 1.076v.666z"
                                      fill="#3E7BFA"
                                    />
                                  </svg>
                                </span>
                                <div
                                  class="Layout--vertical StyledProps--main StyledProps--padding-left-small"
                                >
                                  <p
                                    class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-weight-bold StyledProps--color StyledProps--color-black"
                                    style="--text-line-clamp: 1;"
                                  >
                                    demo-1
                                  </p>
                                  <p
                                    class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-size-small StyledProps--font-weight-light StyledProps--color StyledProps--color-grey600"
                                    style="--text-line-clamp: 1;"
                                  >
                                    common.ID: demo1.harnessSecretManager
                                  </p>
                                </div>
                              </div>
                            </div>
                            <button
                              class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main editBtn Button--with-current-color Button--without-shadow"
                              data-testid="-edit"
                              type="button"
                            >
                              <span
                                class="bp3-button-text"
                              >
                                <span
                                  class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey600"
                                  data-icon="Edit"
                                >
                                  <svg
                                    height="16"
                                    viewBox="0 0 16 16"
                                    width="16"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      clip-rule="evenodd"
                                      d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                                      fill="currentColor"
                                      fill-rule="evenodd"
                                    />
                                    <path
                                      clip-rule="evenodd"
                                      d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                                      fill="currentColor"
                                      fill-rule="evenodd"
                                      opacity="0.3"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                      <div
                        class="StyledProps--font StyledProps--main"
                      />
                    </div>
                    <div
                      class="Collapse--collapse"
                    >
                      <div
                        class="bp3-collapse"
                      >
                        <div
                          aria-hidden="false"
                          class="bp3-collapse-body"
                          style="transform: translateY(-undefinedpx);"
                        />
                      </div>
                    </div>
                  </div>
                  <div
                    class="Collapse--main collapseWrapper"
                  >
                    <div
                      class="StyledProps--font StyledProps--main CollapseHeader--main Collapse--header collapseHeader hideCollapseIcon"
                    >
                      <div
                        class="StyledProps--font StyledProps--main CollapseHeader--leftSection"
                      >
                        <span
                          class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey400"
                          data-icon="main-chevron-right"
                        >
                          <svg
                            height="12"
                            viewBox="0 0 8 14"
                            width="12"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M7.7 6.3l-6-6C1.5.1 1.3 0 1 0 .4 0 0 .4 0 1c0 .3.1.5.3.7L5.6 7 .3 12.3c-.2.2-.3.4-.3.7 0 .6.4 1 1 1 .3 0 .5-.1.7-.3l6-6c.2-.2.3-.4.3-.7s-.1-.5-.3-.7z"
                              fill="#4d5969"
                            />
                          </svg>
                        </span>
                        <div
                          class="collapeHeaderContent"
                        >
                          <div
                            class="Layout--horizontal StyledProps--main item StyledProps--flex StyledProps--flex-alignItems-center StyledProps--flex-justifyContent-space-between"
                          >
                            <div
                              class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main leftInfo"
                            >
                              <span
                                class="bp3-icon StyledProps--main iconCheck"
                                data-icon="pipeline-approval"
                              >
                                <svg
                                  height="14"
                                  viewBox="0 0 19 17"
                                  width="14"
                                  xmlns="http://www.w3.org/2000/svg"
                                >
                                  <path
                                    d="M18.81.669l.006.004.015.013a.5.5 0 01.062.7L6.479 16.375a.5.5 0 01-.709.062.52.52 0 01-.136-.176.523.523 0 01-.126-.095L.132 10.578a.5.5 0 01.005-.698l.017-.018a.5.5 0 01.707-.005l.007.007 5.233 5.367L18.107.735a.5.5 0 01.704-.066z"
                                    fill="currentColor"
                                    fill-rule="evenodd"
                                  />
                                </svg>
                              </span>
                              <div
                                class="Layout--horizontal StyledProps--main StyledProps--flex StyledProps--flex-alignItems-center"
                              >
                                <span
                                  class="bp3-icon StyledProps--main secretIcon"
                                  data-icon="text"
                                >
                                  <svg
                                    fill="none"
                                    height="24"
                                    viewBox="0 0 16 19"
                                    width="24"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      d="M11.585 18.334H4.467v-.667c.371-.135 1.916-1.009 2.356-1.076V1.667H2.947L1.3 5.14H.667c-.051-.338 0-4.469 0-4.807h14.718c0 .338.05 4.452 0 4.807h-.658L13.13 1.667H9.255V16.59c.439.084 1.976.924 2.33 1.076v.666z"
                                      fill="#3E7BFA"
                                    />
                                  </svg>
                                </span>
                                <div
                                  class="Layout--vertical StyledProps--main StyledProps--padding-left-small"
                                >
                                  <p
                                    class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-weight-bold StyledProps--color StyledProps--color-black"
                                    style="--text-line-clamp: 1;"
                                  >
                                    text1
                                  </p>
                                  <p
                                    class="StyledProps--font Text--lineclamp StyledProps--main StyledProps--font-size-small StyledProps--font-weight-light StyledProps--color StyledProps--color-grey600"
                                    style="--text-line-clamp: 1;"
                                  >
                                    common.ID: text1.vault1
                                  </p>
                                </div>
                              </div>
                            </div>
                            <button
                              class="bp3-button bp3-minimal Button--button StyledProps--font StyledProps--main editBtn Button--with-current-color Button--without-shadow"
                              data-testid="-edit"
                              type="button"
                            >
                              <span
                                class="bp3-button-text"
                              >
                                <span
                                  class="bp3-icon StyledProps--main StyledProps--color StyledProps--color-grey600"
                                  data-icon="Edit"
                                >
                                  <svg
                                    height="16"
                                    viewBox="0 0 16 16"
                                    width="16"
                                    xmlns="http://www.w3.org/2000/svg"
                                  >
                                    <path
                                      clip-rule="evenodd"
                                      d="M11.207.485a.5.5 0 01.354-.146 3.5 3.5 0 013.5 3.5.5.5 0 01-.146.353l-11 11a.5.5 0 01-.354.147h-3a.5.5 0 01-.5-.5v-3a.5.5 0 01.146-.354l11-11zm.553.862L1.061 12.046v2.293h2.293l10.7-10.7a2.5 2.5 0 00-2.294-2.292z"
                                      fill="currentColor"
                                      fill-rule="evenodd"
                                    />
                                    <path
                                      clip-rule="evenodd"
                                      d="M11.89 2.646a.5.5 0 010 .708L8.353 6.889a.5.5 0 01-.708-.707l3.536-3.536a.5.5 0 01.707 0z"
                                      fill="currentColor"
                                      fill-rule="evenodd"
                                      opacity="0.3"
                                    />
                                  </svg>
                                </span>
                              </span>
                            </button>
                          </div>
                        </div>
                      </div>
                      <div
                        class="StyledProps--font StyledProps--main"
                      />
                    </div>
                    <div
                      class="Collapse--collapse"
                    >
                      <div
                        class="bp3-collapse"
                      >
                        <div
                          aria-hidden="false"
                          class="bp3-collapse-body"
                          style="transform: translateY(-undefinedpx);"
                        />
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="Layout--horizontal StyledProps--main Pagination--container StyledProps--padding-top-medium StyledProps--padding-bottom-medium StyledProps--flex StyledProps--flex-align-center-center StyledProps--flex-distribution-space-between"
                >
                  <span
                    class="StyledProps--font StyledProps--main StyledProps--inline StyledProps--font-variation-small-semi"
                  >
                    3
                     of 
                    3
                  </span>
                  <div
                    class="Layout--horizontal StyledProps--main"
                  >
                    <button
                      class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--buttonLeft Button--with-current-color Button--withLeftIcon Button--size Button--small"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="bp3-icon bp3-icon-arrow-left StyledProps--main StyledProps--padding-right-xsmall"
                        icon="arrow-left"
                      >
                        <svg
                          data-icon="arrow-left"
                          height="12"
                          viewBox="0 0 16 16"
                          width="12"
                        >
                          <desc>
                            arrow-left
                          </desc>
                          <path
                            d="M13.99 6.99H4.41L7.7 3.7a1.003 1.003 0 00-1.42-1.42l-5 5a1.014 1.014 0 000 1.42l5 5a1.003 1.003 0 001.42-1.42L4.41 8.99H14c.55 0 1-.45 1-1s-.46-1-1.01-1z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                      <span
                        class="bp3-button-text"
                      >
                        Prev
                      </span>
                    </button>
                    <button
                      class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--selected Button--with-current-color"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="bp3-button-text"
                      >
                        1
                      </span>
                    </button>
                    <button
                      class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main Pagination--roundedButton Pagination--buttonRight Button--with-current-color Button--withRightIcon Button--size Button--small"
                      disabled=""
                      tabindex="-1"
                      type="button"
                    >
                      <span
                        class="bp3-button-text"
                      >
                        Next
                      </span>
                      <span
                        class="bp3-icon bp3-icon-arrow-right StyledProps--main"
                        icon="arrow-right"
                      >
                        <svg
                          data-icon="arrow-right"
                          height="12"
                          viewBox="0 0 16 16"
                          width="12"
                        >
                          <desc>
                            arrow-right
                          </desc>
                          <path
                            d="M14.7 7.29l-5-5a.965.965 0 00-.71-.3 1.003 1.003 0 00-.71 1.71l3.29 3.29H1.99c-.55 0-1 .45-1 1s.45 1 1 1h9.59l-3.29 3.29a1.003 1.003 0 001.42 1.42l5-5c.18-.18.29-.43.29-.71s-.12-.52-.3-.7z"
                            fill-rule="evenodd"
                          />
                        </svg>
                      </span>
                    </button>
                  </div>
                  <span>
                    Showing 100 per page
                  </span>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="Layout--horizontal Layout--layout-spacing-medium StyledProps--main StyledProps--padding-top-medium"
        >
          <button
            class="bp3-button bp3-disabled Button--button StyledProps--font StyledProps--main bp3-popover-dismiss Button--with-current-color Button--variation Button--variation-primary"
            disabled=""
            tabindex="-1"
            type="button"
          >
            <span
              class="bp3-button-text"
            >
              entityReference.apply
            </span>
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
